### PR TITLE
Surya/core 10499/secretservice impl

### PR DESCRIPTION
**Pull request by heronhaye**
Addresses https://github.com/keybase/client/issues/16564.

Max wanted me to disable this for nopw since it would be bad if we lost a user's randompw. Not sure if I did it in the cleanest way, open to restructuring the SecretStoreOptions thing. I also disabled automatic upgrades for now, so only new devices or people who log out/log in will have it stored in the keyring. So hopefully we get some useful feedback.

There are generic tests for the secretstore so I didn't add any specifically for secretservice, since it won't work in CI anyway.

Outstanding question: should we add gnome keyring or kwallet to our dependencies? Users of the one we don't pick might be irked though

### CODE REVIEW

**Comment by eli-schwartz** [comment]
> Outstanding question: should we add gnome keyring or kwallet to our dependencies? Users of the one we don't pick might be irked though

At most, you should add an optdepends (pacman) or Suggests (dpkg). Many Desktop Environments already pull in a secretservice frontend for the user anyway.

There is precedent in Arch Linux for packages which optionally depend on e.g. gnome-keyring. See for example https://www.archlinux.org/packages/extra/x86_64/chromium/

File: go/engine/signup.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ func (s *SignupEngine) Run(m libkb.MetaContext) (err error) {
 		return err
 	}
 
-	if err = s.registerDevice(m, s.arg.DeviceName); err != nil {
+	if err = s.registerDevice(m, s.arg.DeviceName, s.arg.GenerateRandomPassphrase); err != nil {
```
Comment (patrickxb) [inline go/engine/signup.go]: What's this random passphrase stuff in signup?

Comment (heronhaye) [reply go/engine/signup.go]: "random passphrase" is the same as "no passphrase". I needed to thread it through to the point where we store it so we can decide to not use the keyring in this case.

Comment (patrickxb) [reply go/engine/signup.go]: But isn't "no passphrase" a separate project?

**Pull request merged**